File: app/src/main/java/com/botijasoftware/utils/collision/CollisionCapsule.java
package com.botijasoftware.utils.collision;


import com.botijasoftware.utils.Vector3;

public class CollisionCapsule extends CollisionVolume {
	private Capsule capsule;
	
	public CollisionCapsule(Vector3 p0, Vector3 p1, float radius) {
		capsule = new Capsule(p0, p1, radius);
	}
	
	public boolean collides(CollisionVolume c) {
		return c.collides(capsule);
	}
	
	public boolean collides(Sphere s) {
		return capsule.collides(s);
	}
	
	public boolean collides(Vector3 v) {
		return capsule.collides(v);
	}
	
	public void setRadius(float r) {
		capsule.setRadius(r);
	}
	
	public float getRadius() {
		return capsule.radius;
	}
	
	public void set(Vector3 v0, Vector3 v1) {
		capsule.set(v0, v1);
	}
	
	public Vector3 getP0() {
		return capsule.getP0();
	}
	
	public Vector3 getP1() {
		return capsule.getP1();
	}
	
}

File: app/src/main/java/com/botijasoftware/utils/collision/CollisionCapsule.kt
package com.botijasoftware.utils.collision


import com.botijasoftware.utils.Vector3

class CollisionCapsule(p0: Vector3, p1: Vector3, radius: Float) : CollisionVolume() {
    private val capsule: Capsule

    init {
        capsule = Capsule(p0, p1, radius)
    }

    override fun collides(c: CollisionVolume): Boolean {
        return c.collides(capsule)
    }

    override fun collides(s: Sphere): Boolean {
        return capsule.collides(s)
    }

    override fun collides(v: Vector3): Boolean {
        return capsule.collides(v)
    }

    override var radius: Float
        get() = capsule.radius
        set(r) {
            capsule.radius = r
        }

    operator fun set(v0: Vector3, v1: Vector3) {
        capsule.set(v0, v1)
    }

    val p0: Vector3
        get() = capsule.p0

    val p1: Vector3
        get() = capsule.p1

}